lib/logstash/event.rb

```diff
@@ -295,11 +295,7 @@ def init_timestamp(o)
 
   public
   def to_hash_with_metadata
-    if @metadata.nil?
-      to_hash
-    else
-      to_hash.merge(METADATA => @metadata)
-    end
+    to_hash.merge(METADATA => @metadata)
```

Comment:
since @metadata can never be nil per https://github.com/elastic/logstash/blob/master/lib/logstash/event.rb#L77 I believe the intent is to not include a @metadata field when it is empty, so it should actually be:
```ruby
@metadata.empty? ? to_hash : to_hash.merge(METADATA => @metadata)
```

no?